backend/app/services/db/rating_manager.py

```diff
@@ -0,0 +1,187 @@
+        cursor.execute(query, values)
+        conn.commit()
+
+    return True
+
+
+def update_rating(rater_name: str, rated_name: str, rating: int):
+    """Updates the information of a user in the database.
+
+    Args:
+        rater_name (str): The username of the rating user.
+        rated_name (str): The username of the user receiving a rating.
+        rating (int): The rating received by the receiving user. Ranges from 1 to 5.
+
+    Returns:
+        True if the function succeeds, else False.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor()
+
+        # Check if users exist
+        existing = get_specific_rating(rater_name, rated_name)
+
+        if not existing:
+            return False
+
+        query = '''
+        UPDATE ratings SET
+        rating = COALESCE(%s, rating)
+        WHERE rater_name = %s AND rated_name = %s
+        '''
+
+        values = (
+            rating,
+            rater_name,
+            rated_name
+        )
+
+        cursor.execute(query, values)
+        conn.commit()
+
+    return True
+
+
+def get_rating(username: str, is_rater: bool = True):
+    """Searches for the ratings of a user in the database. If found, returns all ratings.
+
+    Args:
+        username (str): The username associated with the user.
+        is_rater (bool): If true, returns all outgoing ratings from the user. Else, returns all ratings received.
+
+    Returns:
+        A list of all ratings received or given by the user.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor(dictionary=True)
+
+        # Check if user exists and get the user id
+        user_details = get_user(username, is_username=True)
+
+        if not user_details:
+            return None
+
+        if is_rater:
+            query = "SELECT * FROM ratings WHERE rater_name = %s"
+        else:
+            query = "SELECT * FROM ratings WHERE rated_name = %s"
+
+        cursor.execute(query, (user_details['username'], ))
+        ratings = cursor.fetchall()
+
+    return ratings
+
+
+def get_specific_rating(rater_name: str, rated_name: str):
+    """Searches for a specific rating in the database.
+
+    Args:
+        rater_name (str): The username of the rating user.
+        rated_name (str): The username of the user receiving a rating.
+
+    Returns:
+        A dict with the details of the specific rating.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor(dictionary=True)
+
+        # Check if users exist
+        userA_details = get_user(rater_name, is_username=True)
+        userB_details = get_user(rated_name, is_username=True)
+
+        if not userA_details or not userB_details:
+            return None
+
+        query = "SELECT * FROM ratings WHERE rater_name = %s AND rated_name = %s"
+
+        cursor.execute(query, (rater_name, rated_name,))
+        ratings = cursor.fetchone()
+
+    return ratings
+
+
+def get_avg_rating(username: str):
```

Comment:
Could you turn this into get_rating_details() and have it return also the number of ratings total in a dictionary

Reply:
Also make the changes to reflect on the API

Reply:
Actually, make this get_user_rating(). If you can come up with a better name too, go for it.